client.go

```diff
@@ -0,0 +1,370 @@
+/**
+ * Copyright (c) 2017 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package goovn
+
+import (
+	"sync"
+
+	"github.com/ebay/libovsdb"
+)
+
+type Client interface {
+	// Get logical switch by name
+	LSGet(ls string) ([]*LogicalSwitch, error)
+	// Create ls named SWITCH
+	LSAdd(ls string) (*OvnCommand, error)
+	// Del ls and all its ports
+	LSDel(ls string) (*OvnCommand, error)
+	// Get all logical switches
+	LSList() ([]*LogicalSwitch, error)
+	// Add external_ids to logical switch
+	LSExtIdsAdd(ls string, external_ids map[string]string) (*OvnCommand, error)
+	// Del external_ids from logical_switch
+	LSExtIdsDel(ls string, external_ids map[string]string) (*OvnCommand, error)
+
+	// Add logical port PORT on SWITCH
+	LSPAdd(ls string, lsp string) (*OvnCommand, error)
+	// Delete PORT from its attached switch
+	LSPDel(lsp string) (*OvnCommand, error)
+	// Set addressset per lport
+	LSPSetAddress(lsp string, addresses ...string) (*OvnCommand, error)
+	// Set port security per lport
+	LSPSetPortSecurity(lsp string, security ...string) (*OvnCommand, error)
+	// Get all lport by lswitch
+	LSPList(ls string) ([]*LogicalSwitchPort, error)
+
+	// Add LB to LSW
+	LSLBAdd(ls string, lb string) (*OvnCommand, error)
+	// Delete LB from LSW
+	LSLBDel(ls string, lb string) (*OvnCommand, error)
+	// List Load balancers for a LSW
+	LSLBList(ls string) ([]*LoadBalancer, error)
+
+	// Add ACL
+	ACLAdd(ls, direct, match, action string, priority int, external_ids map[string]string, logflag bool, meter string) (*OvnCommand, error)
+	// Delete acl
+	ACLDel(ls, direct, match string, priority int, external_ids map[string]string) (*OvnCommand, error)
+	// Get all acl by lswitch
+	ACLList(ls string) ([]*ACL, error)
+
+	// Get AS
+	ASGet(name string) ([]*AddressSet, error)
```

Comment:
It doesn't make sense to return a list, because the ovn-nb schema requires that name is unique in the Address_Set table.

Reply:
yes, fixed